tests/test_tools_filter_dom.py

```
@@ -0,0 +1,123 @@
+    assert len(out_data["pages"]) == 1
+
+
+def test_filter_dom_file_not_found(tmp_path):
+    with pytest.raises(FileNotFoundError, match="File not found"):
+        filter_dom(tmp_path / "nonexistent.json", tmp_path / "out.json")
+
+
+def test_filter_dom_invalid_extension(tmp_path):
+    input_file = tmp_path / "input.txt"
+    input_file.touch()
+    with pytest.raises(ValueError, match="Input file must be a .json file"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)